core/src/main/java/tech/tablesaw/api/Row.java

```diff
@@ -4,39 +4,90 @@
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
+import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
-import java.util.TreeMap;
 import tech.tablesaw.columns.Column;
 import tech.tablesaw.table.TableSlice;
 
 public class Row implements Iterator<Row> {
 
+  /**
+   * Wrap Map of column name to Column map and provide helpful error messages to the user when a
+   * column name cannot be found.
+   */
+  private class ColumnMap<T> {
```

Comment:
make this static?

Reply:
In order to provide the context in the error messages ColumnMap needs access to the table and list of columns. If we wanted to make it static we could create all the ColumnMaps in Row's constructor, but it is more concise the way it is. Happy to change it if you like.

As far as inner classes go I think they are fine as long as they are private and no object outside the containing class can hold a reference to the inner class.